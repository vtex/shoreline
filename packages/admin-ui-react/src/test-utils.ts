import type { ReactElement } from 'react'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { createSystem } from './system'

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export { render }
