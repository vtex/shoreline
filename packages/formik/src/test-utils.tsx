import '@testing-library/jest-dom/extend-expect'
import React, { useRef } from 'react'
import { axe, toHaveNoViolations } from 'jest-axe'
import type { RenderOptions } from '@testing-library/react'
import {
  render as baseRender,
  waitFor,
  screen,
  fireEvent,
} from '@testing-library/react'
import { createSystem } from '@vtex/admin-ui-react'
import type { ReactElement } from 'react'
import { renderHook, act } from '@testing-library/react-hooks'

expect.extend(toHaveNoViolations)

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

/**
 * HOC to wrap a component with its state hook
 * @example
 * // Cmp has the useCmp state hook
 * const StatefulCpm = withState(Cmp, () => useCmpState({ isSelected: false }))
 */
function withState(Component: any, useStateHook: () => any) {
  function StatefulComponent(props: any) {
    // Avoids an invalid hook call
    const ref = useRef(useStateHook())

    return <Component state={ref.current} {...props} />
  }

  StatefulComponent.displayName = Component.displayName

  return StatefulComponent
}

function jestMatchMedia() {
  /**
   * ðŸš§ Workaround for window.match media
   * @see https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
   */
  return Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  })
}

export {
  render,
  withState,
  axe,
  jestMatchMedia,
  renderHook,
  act,
  waitFor,
  screen,
  fireEvent,
}
