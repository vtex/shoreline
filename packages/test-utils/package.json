{
  "name": "@vtex/shoreline-test-utils",
  "version": "1.0.32",
  "description": "Shoreline test utilities",
  "main": "./dist/index.mjs",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.mts",
  "repository": "vtex/shoreline",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "dist"
  ],
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "types": "./dist/index.d.mts"
    },
    "./setup": {
      "import": "./dist/setup.mjs",
      "types": "./dist/setup.d.mts"
    }
  },
  "engines": {
    "node": ">=20"
  },
  "scripts": {
    "prebuild": "rm -rf dist",
    "dev": "tsup --watch",
    "build": "npm run prebuild && tsup"
  },
  "peerDependencies": {
    "vitest": "0.x"
  },
  "devDependencies": {
    "vitest": "0.34.6"
  },
  "dependencies": {
    "@testing-library/dom": "9.3.4",
    "@testing-library/jest-dom": "6.4.6",
    "@testing-library/react": "14.3.1",
    "@testing-library/user-event": "14.5.2"
  }
}
