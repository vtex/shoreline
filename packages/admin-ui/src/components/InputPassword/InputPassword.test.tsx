import React from 'react'
import { render, axe } from '../../test-utils'
import { IconPlus } from '@vtex/phosphor-icons'

import { InputPassword } from './index'

describe('InputPassword tests', () => {
  it('should not have a11y violations', async () => {
    const { container } = render(
      <>
        <InputPassword
          criticalText="Error Message"
          tone="critical"
          value=""
          onChange={() => {}}
          label="Label"
          id="text-field-1"
        />
        <InputPassword
          icon={<IconPlus />}
          charLimit={120}
          helperText="Helper Text"
          criticalText="Error Message"
          value=""
          onChange={() => {}}
          label="Label"
          id="text-field-7"
        />
      </>
    )

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
