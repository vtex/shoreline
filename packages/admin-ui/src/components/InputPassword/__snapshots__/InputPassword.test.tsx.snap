// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputPassword tests should match snapshot 1`] = `
<DocumentFragment>
  .emotion-10 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-10 input:focus+label,
.emotion-10 textarea:focus+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-10 input:placeholder-shown:not(:focus)+label,
.emotion-10 textarea:placeholder-shown:not(:focus)+label {
  padding-top: 0.25rem;
}

.emotion-10 input:not(:placeholder-shown)+label,
.emotion-10 textarea:not(:placeholder-shown)+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-7 {
  position: relative;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  left: 12px;
  padding-top: 0.5rem;
  color: #545454;
  margin-bottom: 0.75rem;
  position: absolute;
  -webkit-transform: translate(0, 16px) scale(1);
  -moz-transform: translate(0, 16px) scale(1);
  -ms-transform: translate(0, 16px) scale(1);
  transform: translate(0, 16px) scale(1);
  transform-origin: top left;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

.emotion-6 {
  right: 0rem;
  top: 0.25rem;
  height: 46px;
  padding-right: 0.75rem;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #000000;
}

.emotion-5 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 500;
  font-size: 0.875rem;
  line-height: 1.1428571429;
  letter-spacing: 0.01em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  color: #545454;
  background: transparent;
  padding: 0.5rem 0.75rem;
  height: 2.25rem;
  margin-top: 0.5rem;
}

.emotion-5:hover {
  color: #3B3B3B;
  background: rgba(0, 0, 0, 0.05);
}

.emotion-5:active {
  color: #000000;
  background: rgba(0, 0, 0, 0.1);
}

.emotion-5:disabled {
  background: transparent;
  color: #747474;
}

.emotion-5:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F4F4F4,0rem 0rem 0rem 0.1875rem #B9B9B9;
}

.emotion-5:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 500;
  font-size: 0.875rem;
  line-height: 1.1428571429;
  letter-spacing: 0.01em;
  visibility: visible;
}

.emotion-2 {
  height: 1.25rem;
  width: 1.25rem;
  min-height: 1.25rem;
  min-width: 1.25rem;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-top: 0.25rem;
}

.emotion-0 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  width: 100%;
  height: 48px;
  padding-left: 0.75rem;
  padding-right: 44px;
  background: transparent;
  border: 1px solid #CC3D3D;
  border-radius: 4px;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
  color: #000000;
  outline: 0;
  -webkit-transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  padding-top: 1rem;
}

.emotion-0:hover {
  border: 1px solid #962828;
}

.emotion-0:focus {
  border: 1px solid #782020;
  box-shadow: 0rem 0rem 0rem 0.0625rem #FBF2F2,0rem 0rem 0rem 0.1875rem #F49494;
}

.emotion-0:disabled {
  background: #E6E6E6;
  color: #747474;
}

.emotion-8 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #CC3D3D;
}

<div
    class="emotion-10"
  >
    <div
      class="emotion-7"
    >
      <input
        class="emotion-0"
        id="text-field-1"
        placeholder=" "
        type="password"
        value=""
      />
      <label
        class="emotion-1"
        for="text-field-1"
      >
        Label
      </label>
      <div
        class="emotion-6"
      >
        <button
          aria-label="show text-field-1 content"
          class="emotion-5"
          type="button"
        >
          <div
            class="emotion-4"
          >
            <div
              class="emotion-3"
            >
              <svg
                class="emotion-2"
                fill="currentColor"
                height="1.25rem"
                viewBox="0 0 256 256"
                width="1.25rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="16rem"
                  width="16rem"
                />
                <path
                  d="M128,55.99219C48,55.99219,16,128,16,128s32,71.99219,112,71.99219S240,128,240,128,208,55.99219,128,55.99219Z"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="18"
                />
                <circle
                  cx="128"
                  cy="128.00061"
                  fill="none"
                  r="40"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="18"
                />
              </svg>
            </div>
          </div>
        </button>
      </div>
    </div>
    <div
      class="emotion-9"
    >
      <span
        class="emotion-8"
      >
        Error Message
      </span>
    </div>
  </div>
  .emotion-12 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-12 input:focus+label,
.emotion-12 textarea:focus+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-12 input:placeholder-shown:not(:focus)+label,
.emotion-12 textarea:placeholder-shown:not(:focus)+label {
  padding-top: 0.25rem;
}

.emotion-12 input:not(:placeholder-shown)+label,
.emotion-12 textarea:not(:placeholder-shown)+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  left: 12px;
  padding-top: 0.5rem;
  color: #545454;
  margin-bottom: 0.75rem;
  position: absolute;
  -webkit-transform: translate(0, 16px) scale(1);
  -moz-transform: translate(0, 16px) scale(1);
  -ms-transform: translate(0, 16px) scale(1);
  transform: translate(0, 16px) scale(1);
  transform-origin: top left;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

.emotion-7 {
  right: 0rem;
  top: 0.25rem;
  height: 46px;
  padding-right: 0.75rem;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #000000;
}

.emotion-6 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 500;
  font-size: 0.875rem;
  line-height: 1.1428571429;
  letter-spacing: 0.01em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  color: #545454;
  background: transparent;
  padding: 0.5rem 0.75rem;
  height: 2.25rem;
  margin-top: 0.5rem;
}

.emotion-6:hover {
  color: #3B3B3B;
  background: rgba(0, 0, 0, 0.05);
}

.emotion-6:active {
  color: #000000;
  background: rgba(0, 0, 0, 0.1);
}

.emotion-6:disabled {
  background: transparent;
  color: #747474;
}

.emotion-6:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F4F4F4,0rem 0rem 0rem 0.1875rem #B9B9B9;
}

.emotion-6:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 500;
  font-size: 0.875rem;
  line-height: 1.1428571429;
  letter-spacing: 0.01em;
  visibility: visible;
}

.emotion-3 {
  height: 1.25rem;
  width: 1.25rem;
  min-height: 1.25rem;
  min-width: 1.25rem;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-top: 0.25rem;
}

.emotion-9 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #545454;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  width: 100%;
  height: 48px;
  padding-left: 44px;
  padding-right: 44px;
  background: transparent;
  border: 1px solid #B9B9B9;
  border-radius: 4px;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
  color: #000000;
  outline: 0;
  -webkit-transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  padding-top: 1rem;
}

.emotion-1:hover {
  border: 1px solid #545454;
}

.emotion-1:focus {
  border: 1px solid #3B3B3B;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F4F4F4,0rem 0rem 0rem 0.1875rem #B9B9B9;
}

.emotion-1:disabled {
  background: #E6E6E6;
  color: #747474;
}

.emotion-8 {
  position: relative;
}

.emotion-8 label {
  left: 44px;
}

.emotion-0 {
  height: 20px;
  width: 20px;
  min-height: 20px;
  min-width: 20px;
  color: #545454;
  top: 0.25rem;
  left: 0rem;
  margin-left: 0.75rem;
  margin-right: 0.75rem;
  margin-top: 14px;
  position: absolute;
}

<div
    class="emotion-12"
  >
    <div
      class="emotion-8"
    >
      <svg
        class="emotion-0"
        fill="currentColor"
        height="1.25rem"
        viewBox="0 0 256 256"
        width="1.25rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="16rem"
          width="16rem"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="18"
          x1="40"
          x2="216"
          y1="128"
          y2="128"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="18"
          x1="128"
          x2="128"
          y1="40"
          y2="216"
        />
      </svg>
      <input
        class="emotion-1"
        id="text-field-7"
        maxlength="120"
        placeholder=" "
        type="password"
        value=""
      />
      <label
        class="emotion-2"
        for="text-field-7"
      >
        Label
      </label>
      <div
        class="emotion-7"
      >
        <button
          aria-label="show text-field-7 content"
          class="emotion-6"
          type="button"
        >
          <div
            class="emotion-5"
          >
            <div
              class="emotion-4"
            >
              <svg
                class="emotion-3"
                fill="currentColor"
                height="1.25rem"
                viewBox="0 0 256 256"
                width="1.25rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="16rem"
                  width="16rem"
                />
                <path
                  d="M128,55.99219C48,55.99219,16,128,16,128s32,71.99219,112,71.99219S240,128,240,128,208,55.99219,128,55.99219Z"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="18"
                />
                <circle
                  cx="128"
                  cy="128.00061"
                  fill="none"
                  r="40"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="18"
                />
              </svg>
            </div>
          </div>
        </button>
      </div>
    </div>
    <div
      class="emotion-11"
    >
      <span
        class="emotion-9"
      >
        Helper Text
      </span>
      <span
        class="emotion-9"
      >
        0/120
      </span>
    </div>
  </div>
</DocumentFragment>
`;
